consensus/parlia/fusionfork.go

```diff
@@ -0,0 +1,226 @@
+
+	blockNr := rpc.BlockNumberOrHashWithHash(header.ParentHash, false)
+	toAddress := common.HexToAddress(systemcontracts.StakeHubContract)
+	gas := (hexutil.Uint64)(uint64(math.MaxUint64 / 2))
+
+	var allValidators []common.Address
+	var allVotingPowers []*big.Int
+	for {
+		data, err := p.stakeHubABI.Pack(method, big.NewInt(int64(len(allValidators))), big.NewInt(100))
+		if err != nil {
+			log.Error("Unable to pack tx for getValidatorWithVotingPower", "error", err)
+			return err
+		}
+		msgData := (hexutil.Bytes)(data)
+
+		result, err := p.ethAPI.Call(ctx, ethapi.TransactionArgs{
+			Gas:  &gas,
+			To:   &toAddress,
+			Data: &msgData,
+		}, blockNr, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		var validators []common.Address
+		var votingPowers []*big.Int
+		var totalLength *big.Int
+		if err := p.stakeHubABI.UnpackIntoInterface(&[]interface{}{&validators, &votingPowers, &totalLength}, method, result); err != nil {
+			return err
+		}
+
+		allValidators = append(allValidators, validators...)
+		allVotingPowers = append(allVotingPowers, votingPowers...)
+		if totalLength.Cmp(big.NewInt(int64(len(allValidators)))) == 0 {
+			break
+		}
+	}
+
+	// 2. sort by voting power
+	method = "maxElectedValidators"
+	data, err := p.stakeHubABI.Pack(method)
+	if err != nil {
+		log.Error("Unable to pack tx for maxElectedValidators", "error", err)
+		return err
+	}
+	msgData := (hexutil.Bytes)(data)
+	result, err := p.ethAPI.Call(ctx, ethapi.TransactionArgs{
+		Gas:  &gas,
+		To:   &toAddress,
+		Data: &msgData,
+	}, blockNr, nil, nil)
+	if err != nil {
+		return err
+	}
+
+	var maxElectedValidators *big.Int
+	if err := p.stakeHubABI.UnpackIntoInterface(&maxElectedValidators, method, result); err != nil {
+		return err
+	}
+
+	var validatorHeap ValidatorHeap
+	if maxElectedValidators.Cmp(big.NewInt(int64(len(allValidators)))) == 1 {
+		for i := 0; i < len(allValidators); i++ {
+			if allVotingPowers[i].Cmp(big.NewInt(0)) == 1 {
+				validatorHeap = append(validatorHeap, ValidatorItem{
+					address:     allValidators[i],
+					votingPower: allVotingPowers[i],
+				})
+			}
+		}
+		sort.SliceStable(validatorHeap, validatorHeap.Less)
+	} else {
+		i := 0
+		for len(validatorHeap) < int(maxElectedValidators.Int64()) && i < len(allValidators) {
+			if allVotingPowers[i].Cmp(big.NewInt(0)) == 1 {
+				validatorHeap = append(validatorHeap, ValidatorItem{
+					address:     allValidators[i],
+					votingPower: allVotingPowers[i],
+				})
+			}
+			i++
+		}
+		hp := &validatorHeap
+		heap.Init(hp)
+		for j := i; j < len(allValidators); j++ {
+			if allVotingPowers[j].Cmp(validatorHeap[0].votingPower) == 1 {
+				heap.Pop(hp)
+				heap.Push(hp, ValidatorItem{
+					address:     allValidators[j],
+					votingPower: allVotingPowers[j],
+				})
+			}
+		}
+	}
+
+	length := len(validatorHeap)
+	eligibleValidators := make([]common.Address, length)
+	eligibleVotingPowers := make([]uint64, length)
```

Comment:
Big problem to use uint64 here: it can only represent  (2**64-1)/1e18 = 18.44 BNB. It have to be big.Int.

Reply:
`votingPower` must be a uint64 value. Because it's from BC before and the decimal of coin in BC is 8